### DISCUSSION TITLE
General Issue: Update `...` and `args` to `sources` in admiral-class based functions

### DISCUSSION THREAD

**Discussion by zdz2101** [pharmaverse/admiral#2219] [Ideas]
### Background Information

As part of our efforts to get to a 1.0 version of admiral, we are looking over all functions for consistency and user-friendliness.

Several functions use `...` or `args` to call upon a list of admiral-specific class (usually xxx_source) objects: e.g. `date_source` `dthcaus_source` `records_source` etc. The following functions are affected:

- `call_derivation()`
- `derive_var_dthcaus()`
- `derive_var_extreme_dt()`
- `derive_var_extreme_dtm()`
- `params()`
- `slice_derivation()`
- `derivation_slice()`
- `restrict_derivation()`
- `slice_derivation()`

### Definition of Done

Add `sources` as a function argument and begin the deprecation process of `...` and `args` for the above functions

**Comment by bundfussr** [comment]
I think `param()` should not be changed.

I would wait until we have cleaned up the functions before implementing this issue. In particular `derive_var_dthcaus()`, `derive_var_extreme_dt()`, and `derive_var_extreme_dtm()` are questionable because they are very similar.

**Comment by bms63** [comment]
Converting back to discussion.  I think after the documentation review we should revisit this topic.  @manciniedoardo @bundfussr @ddsjoberg @zdz2101

**Comment by manciniedoardo** [comment]
Decision: add to agenda for next week's meeting and look into background of `...` 

@bms63
